shared/src/utils/validators/get-map-validator.ts

```diff
@@ -0,0 +1,43 @@
+export interface ValidationFunctions {
+    keyValidator: (key: string) => boolean;
+    /**
+     * This validator must return `false` when {@link value} is not a valid object.
+     */
+    valueValidator: (value: unknown) => boolean;
```

Comment:
```suggestion
    // TODO:  `(value: unknown) => value is T` would be better 
    // but makes the rest of the code harder 
    // because type guards always seem to be converted by typescript to boolean when nesting functions.
    valueValidator: (value: unknown) => boolean;
```

Reply:
I chose to actually use the `is` for these functions and use type assertions in the usages. I'd thought that the disadvantages of an additional type cast do not outweigh the advantage of better readability of the functions.